src/server/qgsrequesthandler.h

```diff
@@ -96,33 +96,26 @@ class QgsRequestHandler
     //! @note not available in Python bindings
     virtual void setGetCoverageResponse( QByteArray* ba ) = 0;
 
-    virtual void setDefaultHeaders() {}
-
     //! Set an HTTP header
     virtual void setHeader( const QString &name, const QString &value ) = 0;
 
     //! Remove an HTTP header
-    virtual int removeHeader( const QString &name ) = 0;
-
-    //! Delete all HTTP headers
-    virtual void clearHeaders() = 0;
+    virtual void removeHeader( const QString &name ) = 0;
 
     //! Append the bytestream to response body
     virtual void appendBody( const QByteArray &body ) = 0;
 
-    //! Clears the response body
-    virtual void clearBody() = 0;
```

Comment:
mmm you can remove a header, but you need to know its name, http://qgis.org/api/classQgsRequestHandler.html#a64f07891172cd1c73d760947c531f418
I'd probably like to see a method to return the header keys as a list so we can clear a single one in case we need to.

About the minimalistic approach, I tend to apply it to the code that I'm supposed to not distribute and over which I have full control of the release cycle, but in this case you need to consider that we are designing an API over an existing one, and that we cannot foresee all the possible use cases and I'm pretty sure that if we cut some of the existing functionality somebody will promptly find a use case for it. 
This of course does not mean we should implement all possible corner cases, but something in the middle would probably fit best.

On public APIs I tend to appreciate a lot rich APIs whre the wise designers leave to the client code the possibility to alter whatever you want without breaking the internals of the class.

Back to the point: it's fine for me to drop the clear* methods if there is a way to do that by clearing the individual items in the header, which is desirable in any case, so I'd add a method to retrieve the header "keys".